File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
#         [6,3,4,5,6]
#           p1     p2
        totalSum = sum(nums)
        leftsum = 0
        for i, x in enumerate(nums):
             rightSum=totalSum - leftsum - x
             if leftsum == rightSum:
                return i
             leftsum += x
        return -1

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let totalSum = nums.reduce((a,b)=>a+b,0)
    let leftSum = 0
    for (let [index,value] of nums.entries()){
        if (leftSum== totalSum-leftSum-value){
            return index
        }
        else{
            leftSum += value
        }
    }
    return -1
};